edocumentation-domain/src/main/java/org/dcsa/edocumentation/domain/persistence/state/BookingStateMachineImpl.java

```diff
@@ -0,0 +1,85 @@
+package org.dcsa.edocumentation.domain.persistence.state;
+
+import lombok.Getter;
+import org.dcsa.edocumentation.domain.persistence.entity.enums.BkgDocumentStatus;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.dcsa.edocumentation.domain.persistence.state.BookingStateMachineTables.BOOKING_STATE_TRANSITION_TABLE;
+import static org.dcsa.edocumentation.domain.persistence.state.BookingStateMachineTables.INITIAL_STATE;
+
+public class BookingStateMachineImpl implements BookingStateMachine {
+
+  @Getter
+  BkgDocumentStatus currentStatus;
+
+  BookingStateMachineImpl() {
+    currentStatus = INITIAL_STATE;
+  }
+
+  BookingStateMachineImpl(BkgDocumentStatus currentStatus) {
```

Comment:
What is the intended use of this contructor?

Reply:
When you load the Booking from the database, you need to resume from a previous state.  This constructor is used in that path (via the static constructor in the interface).